lambdas/passport/src/main/java/uk/gov/di/ipv/cri/passport/passport/validation/AuthRequestValidator.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.di.ipv.cri.passport.passport.validation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.di.ipv.cri.passport.library.error.ErrorResponse;
+import uk.gov.di.ipv.cri.passport.library.service.ConfigurationService;
+import uk.gov.di.ipv.cri.passport.library.validation.ValidationResult;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class AuthRequestValidator {
+
+    public static final String CLIENT_ID_PARAM = "client_id";
+    public static final String REDIRECT_URI_PARAM = "redirect_uri";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthRequestValidator.class);
+
+    private final ConfigurationService configurationService;
+
+    public AuthRequestValidator(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    public ValidationResult<ErrorResponse> validateRequest(
+            Map<String, List<String>> queryStringParameters) {
+        if (queryStringParamsMissing(queryStringParameters)) {
+            LOGGER.error("Missing required query parameters for authorisation request");
+            return new ValidationResult<>(false, ErrorResponse.MISSING_QUERY_PARAMETERS);
+        }
+
+        var redirectErrorResult = validateRedirectUrl(queryStringParameters);
+        if (redirectErrorResult.isPresent()) {
+            return new ValidationResult<>(false, redirectErrorResult.get());
+        }
+
+        return ValidationResult.createValidResult();
+    }
+
+    private boolean queryStringParamsMissing(Map<String, List<String>> queryStringParameters) {
+        return Objects.isNull(queryStringParameters) || queryStringParameters.isEmpty();
+    }
+
+    private Optional<ErrorResponse> validateRedirectUrl(
+            Map<String, List<String>> queryStringParameters) {
+        try {
+            String redirectUrl =
+                    getOnlyValueOrThrow(
+                            queryStringParameters.getOrDefault(REDIRECT_URI_PARAM, List.of()));
+            String clientId =
+                    getOnlyValueOrThrow(
+                            queryStringParameters.getOrDefault(CLIENT_ID_PARAM, List.of()));
+            List<String> clientRedirectUrls = configurationService.getClientRedirectUrls(clientId);
+
+            if (!clientRedirectUrls.contains(redirectUrl)) {
+                LOGGER.error("Invalid redirect URL for client_id {}: '{}'", clientId, redirectUrl);
+                return Optional.of(ErrorResponse.INVALID_REDIRECT_URL);
+            }
+            return Optional.empty();
+        } catch (IllegalArgumentException e) {
+            LOGGER.error(e.getMessage());
+            return Optional.of(ErrorResponse.FAILED_TO_PARSE_OAUTH_QUERY_STRING_PARAMETERS);
```

Comment:
It can also be thrown when trying to read the redirect uri from the list of queryParams if there is more than 1 value in it.

I tried to match this up a bit to how its done in core back, but ended up changing a few small things. I think II've actually reverted those changes now though so to match what core-back currently does for this would be to throw a n INVALID_REQUEST_PARAM error which I'm not so sure about, it seems a bit vague to me